sklearn/feature_selection/tests/test_from_model.py

```diff
@@ -375,3 +378,27 @@ def test_allow_nan_tag_comes_from_estimator():
     no_nan_est = NoNaNTag()
     model = SelectFromModel(estimator=no_nan_est)
     assert model._get_tags()['allow_nan'] is False
+
+
```

Comment:
I would parametrize the test in the following manner:


```python
def _pca_importances(pca_estimator):
    return np.abs(pca_estimator.explained_variance_)


@pytest.mark.parametrize(
    "estimator, importance_getter",
    [(make_pipeline(PCA(n_components=4, random_state=0), LogisticRegression()),
      'named_steps.logisticregression.coef_'),
     (PCA(n_components=4, random_state=0), _pca_importances)]
)
def test_select_from_model_importance_getter(estimator, importance_getter):
    # check that we can specify the feature importances attributes
    # Non-regression test for:
    # https://github.com/scikit-learn/scikit-learn/issues/15312
    # https://github.com/scikit-learn/scikit-learn/issues/16192
    model = SelectFromModel(
        estimator, threshold="mean", importance_getter=importance_getter
    )
    model.fit(data, y)
    assert model.transform(data).shape[1] == 1
```

Reply:
This looks more readable. Updated it.
But same doubt here. Do you we can test for enhancements as Non-regression tests? 
Please correct me if I am wrong.